vm/vm.go

```diff
@@ -315,15 +315,15 @@ func (mon *monitor) createReport(defaultError string) *report.Report {
 			Suppressed: report.IsSuppressed(mon.reporter, mon.output),
 		}
 	}
-	start := mon.matchPos + rep.StartPos - beforeContext
+	start := rep.StartPos - beforeContext
 	if start < 0 {
 		start = 0
 	}
-	end := mon.matchPos + rep.EndPos + afterContext
-	if end > len(mon.output) {
-		end = len(mon.output)
+	end := rep.EndPos + afterContext
+	if end > len(rep.Output) {
+		end = len(rep.Output)
 	}
-	rep.Output = mon.output[start:end]
+	rep.Output = rep.Output[start:end]
 	rep.StartPos += mon.matchPos - start
```

Comment:
Does this need `mon.matchPos`? Shouldn't it be `rep.StartPos -= start`?

Reply:
Indeed. Fixed that.